cmd/mailbox-controller/controller.go

```diff
@@ -144,13 +146,18 @@ func (ctl *mbCtl) enqueue(obj any) {
 		logger.V(4).Info("Enqueuing reference due to workspace", "wsName", typed.Name)
 		ctl.queue.Add(typed.Name)
 	case *edgev2alpha1.SyncTarget:
-		mbwsName := ctl.mbwsNameOfSynctarget(typed)
-		if mbwsName == "" {
-			logger.Error(nil, "Failed to construct mailbox workspace name from SyncTarget", "syncTargetName", typed.Name)
-			return
+		mbwsName, err := ctl.mbwsNameOfSynctarget(typed)
+		if err != nil {
+			if err.Error() == errNoSpaceId.Error() {
+				logger.V(4).Info("Enqueuing SyncTarget for later retry", "syncTargetName", typed.Name)
+				ctl.queue.Add(obj)
```

Comment:
You want to enqueue just _references_ to objects, so that successive references to the same object will be coalesced by the queue.

Reply:
Changed